### PR TITLE
Night/Daytime Messaging TDD Fixes

### LINKED ISSUE #36: Night Action Choice Results
Doppelganger, Lone wolf, Seer, Robber and Insomniac must know the result of night actions. Doppelganger will need to perform other actions, but the remaining role actions and results of Doppelganged role actions can be displayed prior to daybreak or in the DayScreen ¯\_(ツ)_/¯ but must be known.

### PR DESCRIPTION
**Pull request by sequestrian**
Added unit tests to verify and fix functionality of `State`'s `nighttimeMessage`, `daytimeMessage` and underlying utility methods.

### CODE REVIEW

File: server/test/State.test.ts [hunk 1]
```diff
@@ -820,6 +820,14 @@ describe("State", () => {
       //--- ends state.distributeRoles()
     });
 
+    describe("for not a werewolf, mason, minion or doppelganger", () => {
+      it("should have empty nigttime messaging", () => {
```
Comment (chodyo) [inline server/test/State.test.ts]: what is "messaging" here? these might be coordinated roles so are we allowing communication? or is it nighttime instructions? guess i'm just trying to figure out how to read this part

**Review by tealidragon — APPROVED** [review]
Nice work. Look at all those tests! ✅

**Comment by tealidragon** [comment]
--no comment

**Comment by chodyo** [comment]
"Nice work!" -> *immediately closes PR* 😅

**Comment by sequestrian** [comment]
Reopening

File: server/test/State.test.ts (hunk 1 above)
Comment (sequestrian) [reply server/test/State.test.ts]: Changed those labels slightly, but these tests are verifying the functionality of `State.nighttimeMessage` which is the messages included on the Night Screen for:
- what role the Doppelganger became
- who, if any, the werewolves are for the minion, or other werewolves for a werewolf
- who the other mason is

Comment (chodyo) [reply server/test/State.test.ts]: right, the server message

File: server/src/State.ts [hunk 2]
```diff
@@ -383,12 +389,13 @@ export class State extends Schema {
         const role = player.role;
         console.debug(`roleID=${JSON.stringify(role)}`);
 
-        switch (role.name) {
+        let roleName = role.doppelganger ? "doppelganger" : role.name;
```
Comment (chodyo) [inline server/src/State.ts:392]: was this added because the doppelganger truly becomes a role?

Comment (chodyo) [reply server/src/State.ts:392]: and/or what is `role.name` when `role.doppelganger` is set

Comment (sequestrian) [reply server/src/State.ts:392]: That's correct. With the current architecture, it made sense to have something to indicate the doppelganged role while still representing to the game's logic that a player is the correct role during processing. See `distributeDoppelsRole` (`State` line 251) for more information on how the doppelganged role is assigned.

**Pull request merged**